File: programs/rvalues/aggregates/struct/distance.c
struct Point {
    int x;
    int y;
};


void main() {
    struct Point p = { x: 10, y: 20 };
    p.x = 3;
    p.y = 4;
    int result = distance(p);
}

int distance(struct Point p) {
    return p.x * p.x + p.y * p.y;
}

File: programs/rvalues/aggregates/struct/distance.py
from dataclasses import dataclass

@dataclass
class Point:
    x: int
    y: int


def distance(p: Point) -> int:
    return p.x * p.x + p.y * p.y


def main():
    p = Point(1, 2)
    result = distance(p)
    print(result)


# if __name__ == "__main__":
#     main()
